examples/daal/cpp_sycl/source/logistic_regression/logistic_regression_online.h

```diff
@@ -0,0 +1,265 @@
+
+        checkParametersInitialized();
+
+        _solver->input.set(solver::inputArgument, _workPoint);
+        _solver->compute();
+
+        auto partialResult = _solver->getResult();
+        _workPoint = partialResult->get(solver::minimum);
+    }
+
+    // Creates final beta parameter values and resets internal algorithm state to default values
+    void finalizeCompute()
+    {
+        checkParametersInitialized();
+
+        daal_dm::BlockDescriptor<Fptype> bBetas;
+        _workPoint->getBlockOfRows(0, _nBetas*_nBetaCols, daal_dm::readOnly, bBetas);
+        if (_isIntercept)
+        {
+            _finalBetas = daal_dm::SyclHomogenNumericTable<Fptype>::create(bBetas.getBuffer(), _nBetaCols, _nBetas);
+        }
+        else
+        {
+            _finalBetas = daal_dm::SyclHomogenNumericTable<Fptype>::create(_nFeatures, _nBetas, daal_dm::NumericTableIface::doAllocate);
+            daal_dm::BlockDescriptor<Fptype> bFinalBetas;
+            _finalBetas->getBlockOfRows(0, _nBetas, daal_dm::writeOnly, bFinalBetas);
+
+            auto src = bBetas.getBlockPtr();
+            auto dst = bFinalBetas.getBlockPtr();
+
+            for (size_t i = 0; i < _nBetas; i++)
+            {
+                for (size_t j = 0; j < _nFeatures; j++)
+                {
+                    dst[i*_nFeatures + j] = src[i*_nBetaCols + j + 1];
+                }
+            }
+
+            _finalBetas->releaseBlockOfRows(bFinalBetas);
+        }
+        _workPoint->releaseBlockOfRows(bBetas);
+
+        setDefaultState(_nClasses, _nBetas, _nBetaCols);
+    }
+
+    // Provides beta coefficients from the last iteration
+    auto getPartialModel() const
+    {
+        return _workPoint;
+    }
+
+    // Provides final model
+    auto getModel() const
+    {
+        if (!_finalBetas)
+            throw std::runtime_error("Final parameters are not calculated: call finalizeCompute()");
+
+        daal::algorithms::logistic_regression::ModelBuilder<Fptype> mBuilder (_nFeatures, _nClasses);
+        mBuilder.setBeta(_finalBetas);
+        return mBuilder.getModel();
+    }
+
+private:
+    void setDefaultState(size_t nClasses, size_t nBetas, size_t nBetaCols)
+    {
+        _learningRateTable = daal_dm::SyclHomogenNumericTable<Fptype>::create(1, 1, daal_dm::NumericTable::doAllocate, 1.0);
+        _workPoint  = daal_dm::SyclHomogenNumericTable<Fptype>::create(1, nBetas*nBetaCols, daal_dm::NumericTable::doAllocate, 0.0);
+
+        if (nClasses != 2)
+        {
+            daal_dm::BlockDescriptor<Fptype> bd;
+            _workPoint->getBlockOfRows(0, nBetas*nBetaCols, daal_dm::writeOnly, bd);
+            auto ptr = bd.getBlockPtr();
+
+            for (size_t i = 0; i < bd.getNumberOfRows(); i += nBetaCols)
+            {
+                ptr[i] = Fptype(1e-3);
+            }
+
+            _workPoint->releaseBlockOfRows(bd);
+        }
+    }
+
+    void checkParameterDomains(size_t nClasses,
```

Comment:
[minor] I think this is superfluous - we are not writing a full-product class/algorithm, but just an simple example, otherwise the checks will not be enough

Reply:
Why do you think these checks are not enough?